### PR TITLE
launcher: support enabling extension loading (#2594)

### LINKED ISSUE #2594: Launcher: Export a new option to toggle `disable-extensions`
Nowadays, `disable-extensions` is a default flag that doesn't allow me have a test for extension app. It might be useful that user can turn on / off the flag vi an option from outside when to test a chrome extension with lighthouse/chrome-launcher. I usually use a chrome while extension testing like this

```
/Applications/Chromium.app/Contents/MacOS/Chromium --app=file:///chrome-extension/extended/test/index.html --system-developer-mode --load-extensi
on=./test/extension
```

### PR DESCRIPTION
**Pull request by ragingwind**
fixes #2594

I've tested flag status by mocha and running test by [chrome-launcher-cli@extension-enable](https://github.com/ragingwind/chrome-launcher-cli/tree/extension-enable) in manually:

```
chrome --app=file:///Users/ragingwind/Workspace/chrome-extension/extended/test/index.html --system-developer-mode --load-ex
tension=/Users/ragingwind/Workspace/chrome-extension/extended/test/extension --enable-extensions
```

### CODE REVIEW

File: chrome-launcher/README.md [hunk 1]
```diff
@@ -56,6 +56,10 @@ npm install chrome-launcher
   // (optional) Logging level: verbose, info, error, silent
   // Default: 'info'
   logLevel: string;
+
+  // (optional) Enabling extension loading with --load-extension flag
```
Comment (brendankenny) [inline chrome-launcher/README.md]: `Enabling extension loading with --enable-extensions flag`?

Comment (ragingwind) [inline chrome-launcher/README.md]: done

File: chrome-launcher/README.md [hunk 2]
```diff
@@ -56,6 +56,10 @@ npm install chrome-launcher
   // (optional) Logging level: verbose, info, error, silent
   // Default: 'info'
   logLevel: string;
+
+  // (optional) Enabling extension loading with --enable-extensions flag
```
Comment (brendankenny) [inline chrome-launcher/README.md]: `// (optional) Enable extension loading`

File: chrome-launcher/chrome-launcher.ts [hunk 3]
```diff
@@ -94,14 +96,15 @@ export class Launcher {
...
+    this.enableExtensions = defaults(this.opts.enableExtensions, false);
   }
 
   private get flags() {
     const flags = DEFAULT_FLAGS.concat([
       `--remote-debugging-port=${this.port}`,
       // Place Chrome profile in a custom location we'll rm -rf later
       `--user-data-dir=${this.userDataDir}`
-    ]);
+    ]).filter(flag => this.enableExtensions ? flag !== '--disable-extensions' : true);
```
Comment (brendankenny) [inline chrome-launcher/chrome-launcher.ts]: this might be clearer as a separate conditional rather than nested in the `flags` declaration. e.g.

```js
if (this.enableExtensions) {
  flags.filter(flag => flag !== '--disable-extensions');
}
```

File: chrome-launcher/test/chrome-launcher-test.ts [hunk 4]
```diff
@@ -102,4 +102,24 @@ describe('Launcher', () => {
     assert.strictEqual(pid, chromeInstance.pid);
     await chromeInstance.kill();
   });
+
+  it('launching flags without --disable-extensions', async () => {
```
Comment (brendankenny) [inline chrome-launcher/test/chrome-launcher-test.ts]: maybe something like `it('removes --disable-extensions from flags on enableExtensions', async () => {`?

File: chrome-launcher/test/chrome-launcher-test.ts [hunk 5]
```diff
@@ -102,4 +102,24 @@ describe('Launcher', () => {
...
+    chromeInstance.prepare();
+
+    try {
+      await chromeInstance.launch();
+    } catch (err) {
+      return Promise.reject(err);
+    }
+
+    const chromeFlags = spawnStub.getCall(0).args[1] as string[];
+    assert.ok(chromeFlags.indexOf('--disable-extensions') < 0)
```
Comment (brendankenny) [inline chrome-launcher/test/chrome-launcher-test.ts]: `assert.ok(!chromeFlags.includes('--disable-extensions'));`

**Comment by ragingwind** [comment]
I will update it those of issues.

File: chrome-launcher/chrome-launcher.ts (hunk 3 above)
Comment (ragingwind) [inline chrome-launcher/chrome-launcher.ts]: @brendankenny Using a separate conditional statement it leads `const flags` to change into `let flags`. is that ok?  Or we can use splice instead of filter. 

```
let flags = ...
if (this.enableExtensions) {
   flags = flags.filter(flag => flag !== '--disable-extensions');
}
```

File: chrome-launcher/test/chrome-launcher-test.ts (hunk 5 above)
Comment (ragingwind) [inline chrome-launcher/test/chrome-launcher-test.ts]: I've got an error, `test/chrome-launcher-test.ts (124,28): Property 'includes' does not exist on type 'string[]'. (2339)` To solve the error, we need to change target config in tsconfig.json from 'es6' to 'es2016'

File: chrome-launcher/chrome-launcher.ts (hunk 3 above)
Comment (brendankenny) [inline chrome-launcher/chrome-launcher.ts]: whoops, yeah, good point. I'd personally prefer `let`/`filter` over `splice`, but that's purely my own bias. Pick whichever one you fine more readable :)

File: chrome-launcher/test/chrome-launcher-test.ts (hunk 5 above)
Comment (brendankenny) [inline chrome-launcher/test/chrome-launcher-test.ts]: > To solve the error, we need to change target config in tsconfig.json from 'es6' to 'es2016'

gah. I would say switch back to `indexOf`, it's the fast fix we can land now and leave switching the target to a different PR.

(@paulirish's 👍 may mean he supports switching over to `es2016` now, so we'll let him speak up if that's the case :)

Comment (paulirish) [inline chrome-launcher/test/chrome-launcher-test.ts]: yeah i think changing to es2016 seems fine.

File: chrome-launcher/chrome-launcher.ts (hunk 3 above)
Comment (ragingwind) [inline chrome-launcher/chrome-launcher.ts]: `let / filter` 👍

File: chrome-launcher/test/chrome-launcher-test.ts (hunk 5 above)
Comment (ragingwind) [inline chrome-launcher/test/chrome-launcher-test.ts]: I will update it to es2016 in this PR

Comment (brendankenny) [inline chrome-launcher/test/chrome-launcher-test.ts]: fyi @samccone :)

**Comment by ragingwind** [comment]
@brendankenny Passed travis but appveyor returns error but I have no idea. Any comments for?

**Comment by brendankenny** [comment]
for some reason only appveyor has been thinking that chromestatus's homepage loads offline. It's a weird bug, but unaffected by this PR

**Comment by ragingwind** [comment]
@brendankenny Thanks for your reviewing and advising

**Comment by hrgui** [comment]
Just wondering - when will a new version of chrome-launcher be released via npm?

Just came across a scenario where I need to launch a custom extension via chrome-launcher, only to find that --disable-extensions is passed in hardcoded, but luckily I found this PR before making an issue.

**Comment by paulirish** [comment]
@hrgui your wish is my command. 🤠 
i just shipped 0.4.0 of chrome-launcher which includes this PR. 🌵
